code/game/objects/machinery/miner.dm

```diff
@@ -0,0 +1,179 @@
+			"<span class='notice'>You fumble around figuring out [src]'s tubing and plating.</span>")
+			var/fumbling_time = 10 SECONDS - 2 SECONDS * user.skills.getRating("engineer")
+			if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+				return FALSE
+		playsound(loc, 'sound/items/ratchet.ogg', 25, TRUE)
+		user.visible_message("<span class='notice'>[user] starts repairing [src]'s tubing and plating.</span>",
+		"<span class='notice'>You start repairing [src]'s tubing and plating.</span>")
+		if(!do_after(user, 150, TRUE, src, BUSY_ICON_BUILD) && miner_status == MINER_SMALL_DAMAGE)
+			return FALSE
+		playsound(loc, 'sound/items/ratchet.ogg', 25, TRUE)
+		miner_status = MINER_RUNNING
+		user.visible_message("<span class='notice'>[user] repairs [src]'s tubing and plating.</span>",
+		"<span class='notice'>You repair [src]'s tubing and plating.</span>")
+		start_processing()
+		update_icon()
+		return TRUE
+
+/obj/machinery/miner/examine(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	switch(miner_status)
+		if(MINER_DESTROYED)
+			to_chat(user, "<span class='info'>It's heavily damaged, and you can see internal workings.</span>\n<span class='info'>Use a blowtorch, then wirecutters, then a wrench to repair it.</span>")
+		if(MINER_MEDIUM_DAMAGE)
+			to_chat(user, "<span class='info'>It's damaged, and there are broken wires hanging out.</span>\n<span class='info'>Use wirecutters, then wrench to repair it.</span>")
+		if(MINER_SMALL_DAMAGE)
+			to_chat(user, "<span class='info'>It's lightly damaged, and you can see some dents and loose piping.</span>\n<span class='info'>Use a wrench to repair it.</span>")
+		if(MINER_RUNNING)
+			to_chat(user, "<span class='info'>[src]'s internal storage currently has [stored_mineral] sheets stored.</span>")
+
+/obj/machinery/miner/attack_hand(mob/living/user)
+	var/mob/living/L = user
+	if(miner_status != MINER_RUNNING)
+		to_chat(L, "<span class='warning'>[src] is damaged!</span>")
+		return
+	if(stored_mineral == 0)
+		to_chat(L, "<span class='warning'>[src]'s internal storage currently has no minerals stored!</span>")
+		return
+	if(stored_mineral >= 50)	//We shouldnt get more, but just in case
+		start_processing()
+	new mineral_produced(L.loc, stored_mineral)
+	stored_mineral = 0
+
+/obj/machinery/miner/process()
+	if(!(miner_status == MINER_RUNNING))
+		stop_processing()
+		return
+	if(add_tick >= required_ticks && stored_mineral <= 49)	//make one phoron per 60 ticks, or 3 seconds.
+		stored_mineral += 1			
+		add_tick = 0
+	else if(stored_mineral >= 50)
+		stop_processing()
+	else
+		add_tick += 1
+
+/obj/machinery/miner/attack_alien(mob/living/carbon/xenomorph/xeno_attacker)
+	xeno_attacker.do_attack_animation(src, ATTACK_EFFECT_CLAW)
+	xeno_attacker.visible_message("<span class='danger'>[xeno_attacker] slashes \the [src]!</span>", \
+	"<span class='danger'>We slash \the [src]!</span>", null, 5)
+	playsound(loc, "alien_claw_metal", 25, 1)
+	set_miner_status()

```

Comment:
I'd personally prefer this was a factor of the objects health instead of stages like this.
3 hits from a xeno can happen very fast and that can't be tuned.

If this was instead worked out at health breakpoints then you can say 
0 destroyed
1-33 heavily damaged
33-66 damaged
66-99 slighty damaged
100 good continue

Reply:
An example could be like
```
	var/health_percent = round((max_integrity / obj_integrity) * 100)
	switch(health_percent)
		if(-INFINITY to 0)
			//destroyed
		if(1 to 33)
			// heavy
		if(34 to 66)
			// damaged
		if(67 to 99)
			// slighty
		if(100 to INFINITY)
			// working
```